### PR TITLE
Miscellaneous tidying in HOG code

### PR DESCRIPTION
**Pull request by vnavkal**
## Description
Here are a few changes to make the HOG code cleaner.  This is not expected to change its behavior.
* Remove some trailing whitespace
* Remove unused local variable `eps`
* Remove sequential square root and square

@soupault perhaps you could be one of the reviewers for this?  It looks like you might be most familiar with this code.

## Checklist
[It's fine to submit PRs which are a work in progress! But before they are merged, all PRs should provide:]
- [ ] Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)
- [ ] [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- [ ] Gallery example in `./doc/examples` (new features only)
- [ ] Unit tests

[For detailed information on these and other aspects see [scikit-image contribution guidelines](http://scikit-image.org/docs/dev/contribute.html)]

## References
[If this is a bug-fix or enhancement, it closes issue # ]
[If this is a new feature, it implements the following paper: ]

### CODE REVIEW

File: skimage/feature/_hog.py [hunk 1]
```diff
@@ -5,17 +5,17 @@
...
 
-def _hog_normalize_block(block, method, eps=1e-5):
+def _hog_normalize_block(block, method, eps):
     if method == 'L1':
         out = block / (np.sum(np.abs(block)) + eps)
     elif method == 'L1-sqrt':
         out = np.sqrt(block / (np.sum(np.abs(block)) + eps))
     elif method == 'L2':
-        out = block / np.sqrt(np.sqrt(np.sum(block ** 2)) ** 2 + eps ** 2)
+        out = block / np.sqrt(np.sum(block ** 2) + eps ** 2)
```
Comment (soupault) [inline skimage/feature/_hog.py]: These changes do not seem correct to me. Take a look at https://en.wikipedia.org/wiki/Histogram_of_oriented_gradients#Block_normalization .

File: skimage/feature/_hog.py [hunk 2]
```diff
@@ -57,7 +57,7 @@ def hog(image, orientations=9, pixels_per_cell=(8, 8), cells_per_block=(3, 3),
         ``L2-Hys``
            Normalization using L2-norm, followed by limiting the
            maximum values to 0.2 (`Hys` stands for `hysteresis`) and
-           renormalization using L2-norm.           
+           renormalization using L2-norm.
```
Comment (soupault) [inline skimage/feature/_hog.py]: I'd keep only this...

File: skimage/feature/_hog.py [hunk 3]
```diff
@@ -230,12 +230,11 @@ def hog(image, orientations=9, pixels_per_cell=(8, 8), cells_per_block=(3, 3),
     normalized_blocks = np.zeros((n_blocksy, n_blocksx,
                                   by, bx, orientations))
 
-    eps = 1e-8
```
Comment (soupault) [inline skimage/feature/_hog.py]: and this.

Comment (vnavkal) [inline skimage/feature/_hog.py]: ah I see, you prefer for `eps` to be set as a default parameter of `_hog_normalize_block`?  I'll add the default value back and not specify a value for `eps` in the method call below.

File: skimage/feature/_hog.py (hunk 1 above)
Comment (vnavkal) [inline skimage/feature/_hog.py]: hmm, sorry, am I missing something silly?  These expressions should be equivalent: any nonnegative number equals the square of its square root.

Comment (soupault) [inline skimage/feature/_hog.py]: @vnavkal no, you are right, they are equivalent. I decided to follow the formulas exactly, but this way it should be faster.

**Pull request merged**